src/pkg/projects/Microsoft.NETCore.App/Microsoft.NETCore.App.pkgproj

```diff
@@ -80,6 +80,42 @@
                               PreferredPackages="$(Id);@(RuntimeDependency)" />
   </Target>
 
+  <UsingTask TaskName="GetVersionFromAssembly" AssemblyFile="$(LocalBuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.Local.dll"/>
+
+  <Target Name="GenerateHashVersionsFile" BeforeTargets="GetPackageFiles">
+    <PropertyGroup>
+      <_SystemRuntimePath Condition="$([System.String]::new('%(File.Identity)').ToLowerInvariant().Contains('system.runtime.dll'))">%(File.Identity)</_SystemRuntimePath>
+      <_SystemPrivateCoreLibPath Condition="$([System.String]::new('%(File.Identity)').ToLowerInvariant().Contains('system.private.corelib.dll'))">%(File.Identity)</_SystemPrivateCoreLibPath>
+      <_SkipCoreClrHash Condition="'$(_SystemPrivateCoreLibPath)' == ''">true</_SkipCoreClrHash>
+    </PropertyGroup>
+
+    <GetVersionFromAssembly PathToAssembly="$(_SystemRuntimePath)">
+      <Output TaskParameter="AssemblyHash" PropertyName="CoreFxHash" />
+    </GetVersionFromAssembly>
+    <GetVersionFromAssembly PathToAssembly="$(_SystemPrivateCoreLibPath)" Condition="'$(_SkipCoreClrHash)' != 'true'" >
+      <Output TaskParameter="AssemblyHash" PropertyName="CoreClrHash" />
+    </GetVersionFromAssembly>
+
+    <Message Importance="Low" Text="The hash of corefx is $(CoreFxHash) taken from $(_SystemRuntimePath)" />
+    <Message Importance="Low" Text="The hash of coreclr is $(CoreClrHash) taken from $(_SystemPrivateCoreLibPath)" />
+
+    <ItemGroup>
+      <_VersionsFileLines Include="CoreSetup $(LatestCommit)" />
```

Comment:
nit but I would make the names for these commits match the repo names so lowercase and a '-' for core-setup.